packages/flex-plugin-scripts/src/config/devServer/pluginServer.ts

```diff
@@ -1,22 +1,40 @@
-import { logger, paths } from 'flex-dev-utils';
+import { env, logger, FlexPluginError } from 'flex-dev-utils';
 import { Request, Response } from 'express-serve-static-core';
-import { readFileSync } from 'flex-dev-utils/dist/fs';
+import { readPluginsJson } from 'flex-dev-utils/dist/fs';
 import { Configuration } from 'webpack-dev-server';
 import https from 'https';
+import { UserInputPlugin } from '../../scripts/start';
+import { isHTTPS } from 'flex-dev-utils/dist/env';
 
 export interface Plugin {
-  src: string;
+  phase: number;
   name: string;
-  enabled: boolean;
-  remote?: boolean;
+  src: string;
 }
 
 /**
- * Returns local plugins from  public/plugins.json
+ * Returns local plugins from  cli/plugins.json
  * @private
  */
 /* istanbul ignore next */
-export const _getLocalPlugins = () => JSON.parse(readFileSync(paths().app.pluginsJsonPath)) as Plugin[];
+export const _getLocalPlugins = (inputPlugins: UserInputPlugin[]) => {
+  const config = readPluginsJson();
+
+  return inputPlugins.map((plugin) => {
+    const findPlugin = config.plugins.find((p) => p.name === plugin.name);
+    if (findPlugin) {
+      const protocol = 'http' + (env.isHTTPS() ? 's' : '') + '://';
```

Comment:
Move this protocol to outside of the `.map` - it does not need to be created on every loop